ERCS/erc-7818.md

```diff
@@ -0,0 +1,162 @@
+
+- Bonds or financial instruments with defined maturity dates
+- Time-constrained assets within gaming ecosystems
+- Next-gen loyalty programs incorporating expiring rewards or points
+- Prepaid credits for utilities or services (e.g., cashback, data packages, fuel, computing resources) that expire if not used within a specified time frame
+- Postpaid telecom data package allocations that expire at the end of the billing cycle, motivating users to utilize their data before it resets
+- Tokenized e-Money for a closed-loop ecosystem, such as transportation, food court, and retail payments
+
+## Specification
+
+The keywords “MUST”, “MUST NOT”, “REQUIRED”, “SHALL”, “SHALL NOT”, “SHOULD”, “SHOULD NOT”, “RECOMMENDED”, “MAY”, and “OPTIONAL” in this document are to be interpreted as described in RFC 2119.
+
+Implementers of this **REQUIRED** inheritance [ERC-20](./eip-20.md) interface and **MUST** have all of the following functions and all function behavior **MUST** meet the specification.
+
+```solidity
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity >=0.8.0 <0.9.0;
+
+/**
+ * @title ERC-7818: Expirable ERC20
+ * @dev Interface for creating expirable ERC20 tokens.
+ */
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+interface IERC7818 is IERC20 {
+
+    /**
+     * @dev Retrieves the balance of a specific `identifier` owned by an account.
+     * @param account The address of the account.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
+     * @return uint256 The balance of the specified `identifier`.
+     * @notice `identifier` "MUST" represent a unique identifier, and its meaning "SHOULD"
+     * align with how contract maintain the `epoch` in the implementing contract.
+     */
+    function balanceOf(
+        address account,
+        uint256 identifier
+    ) external view returns (uint256);
+
+    /**
+     * @dev Retrieves the current epoch of the contract.
+     * @return uint256 The current epoch of the token contract,
+     * often used for determining active/expired states.
+     */
+    function epoch() external view returns (uint256);
+
+    /**
+     * @dev Retrieves the duration a token remains valid.
+     * @return uint256 The validity duration.
+     * @notice `duration` "MUST" specify the token's validity period.
+     * The implementing contract "SHOULD" clearly document,
+     * whether the unit is blocks or time in seconds.
+     */
+    function duration() external view returns (uint256);
+
+    /**
+     * @dev Checks whether a specific `identifier` is expired.
+     * @param identifier The Identifier "MAY" represents an epoch, round, period, or token identifier.
```

Comment:
So since you have a function `epoch`, does it make sense to also include "round" / "period" as well? Seems like the standard explicitly names it "epoch".